### PR TITLE
Added portworx support changes into elastic-2.7.0-6.8.1-stable

### PR DESCRIPTION
**Pull request by tusharrautOO7**

### CODE REVIEW

**Comment by tusharrautOO7** [comment]
@disrani-px

File: frameworks/elastic/build.gradle [hunk 1]
```diff
@@ -9,21 +9,18 @@ repositories {
...
     mavenCentral()
     maven {
-        url "http://downloads.mesosphere.com/maven/"
+        url "https://s3-us-west-1.amazonaws.com/px-dcos/maven/"
     }
     maven {
-        url "http://downloads.mesosphere.com/maven-snapshot/"
-    }
-    maven {
-        url 'http://repository.apache.org/content/repositories/snapshots/'
```
Comment (piyush-nimbalkar) [inline frameworks/elastic/build.gradle:18]: Is this not needed for other non-sdk packages? Or is just pulling from maven central?

File: frameworks/elastic/universe/resource.json [hunk 2]
```diff
@@ -3,7 +3,8 @@
     "uris": {
       "jre-tar-gz": "{{jre-url}}",
       "libmesos-bundle-tar-gz": "{{libmesos-bundle-url}}",
-      "bootstrap-zip": "{{artifact-dir}}/bootstrap.zip",
+      "bootstrap-zip": "https://s3-us-west-1.amazonaws.com/px-dcos/dcos-commons/artifacts/0.40.5-1.3.4/bootstrap.zip",
+      "executor-zip": "https://s3-us-west-1.amazonaws.com/px-dcos/dcos-commons/artifacts/0.40.5-1.3.4/executor.zip",
```
Comment (piyush-nimbalkar) [inline frameworks/elastic/universe/resource.json]: Does this build the CLIs for the elastic as well? Can you see it getting built in the output and then being uploaded?
I see the build.sh does generate CLIs and bootstrap.zip, but want to make sure.

File: frameworks/elastic/universe/package.json [hunk 3]
```diff
@@ -1,18 +1,28 @@
...
+  ],
   "minDcosReleaseVersion": "1.11",
-  "name": "elastic",
+  "name": "portworx-elastic",
   "version": "{{package-version}}",
-  "maintainer": "support@mesosphere.io",
-  "description": "Elasticsearch 6, and X-Pack",
+  "maintainer": "support@portworx.com",
+  "description": "Elasticsearch 6.8.1, and X-Pack",
   "selected": true,
```
Comment (piyush-nimbalkar) [inline frameworks/elastic/universe/package.json]: Selected should be false for our framework.

File: frameworks/elastic/universe/package.json [hunk 4]
```diff
@@ -1,18 +1,28 @@
...
+  "tags": [
+    "elastic",
+    "elasticsearch",
+    "kibana",
+    "x-pack",
+    "portworx"
+  ],
+  "preInstallNotes": "This DC/OS Service is currently a beta candidate undergoing testing as part of a formal beta test program.\n\nThere may be bugs, incomplete features, incorrect documentation, or other discrepancies.\n\nDefault configuration requires 3 agent nodes each with: CPU: 4.0 | Memory: 9216MB | Disk: 13500MB\n\nMore specifically, each instance type requires:\n\nMaster node: 3 instances | 1.0 CPU | 2048 MB MEM | 1 2000 MB Disk\n\nData node: 2 instances | 1.0 CPU | 4096 MB MEM | 1 10000 MB Disk\n\nCoordinator node: 1 instance | 1.0 CPU | 2048 MB MEM | 1 1000 MB Disk\n\nIngest node: No instances by default | 0.5 CPU | 2048 MB MEM | 1 2000 MB Disk\n\nPortworx should be installed on the Private Agents to be able to provision volumes.\n\nContact Mesosphere before deploying this beta candidate service. Product support is available to approved participants in the beta test program.",
```
Comment (piyush-nimbalkar) [inline frameworks/elastic/universe/package.json]: This is no more a beta candidate, right?
We should just use their current description as is and add Portworx requirement at the end of it.

File: frameworks/elastic/universe/package.json [hunk 5]
```diff
@@ -1,18 +1,28 @@
...
+  "postInstallNotes": "The DC/OS Elastic service is being installed!\n\n\tDocumentation: https://docs.portworx.com/scheduler/mesosphere-dcos/elasticsearch.html",
+  "postUninstallNotes": "The DC/OS Elastic service is being uninstalled\nPlease follow the instructions at https://docs.portworx.com/scheduler/mesosphere-dcos/framework_cleanup.html to remove any persistent state if required.",
```
Comment (piyush-nimbalkar) [inline frameworks/elastic/universe/package.json]: Looks like this link is not valid anymore. We should start using - https://docs.portworx.com/install-with-other/dcos/operate-and-maintain/framework-cleanup

File: frameworks/elastic/build.gradle [hunk 6]
```diff
@@ -9,21 +9,18 @@ repositories {
...
+        url "https://s3-us-west-1.amazonaws.com/px-dcos/maven-snapshot/"
     }
 }
 
 ext {
     junitVer = "4.12"
     systemRulesVer = "1.16.0"
     mockitoVer = "2.10.0"
-    dcosSDKVer = "0.56.2"
+    dcosSDKVer = "0.40.5-1.3.4"
```
Comment (piyush-nimbalkar) [inline frameworks/elastic/build.gradle]: Can we use the latest here?
0.40.5-1.3.7

File: frameworks/elastic/universe/resource.json (hunk 2 above)
Comment (tusharrautOO7) [reply frameworks/elastic/universe/resource.json]: It uploads all cli and bootstrap.zip

File: frameworks/elastic/universe/package.json (hunk 5 above)
Comment (tusharrautOO7) [reply frameworks/elastic/universe/package.json]: Modified the link.

File: frameworks/elastic/universe/package.json (hunk 3 above)
Comment (tusharrautOO7) [reply frameworks/elastic/universe/package.json]: Changed.

File: frameworks/elastic/build.gradle (hunk 6 above)
Comment (tusharrautOO7) [reply frameworks/elastic/build.gradle]: Changed.

File: frameworks/elastic/build.gradle (hunk 1 above)
Comment (tusharrautOO7) [reply frameworks/elastic/build.gradle:18]: Added repository path and build the package.

**Review by piyush-nimbalkar — APPROVED** [review]

File: frameworks/elastic/universe/package.json (hunk 4 above)
Comment (tusharrautOO7) [reply frameworks/elastic/universe/package.json]: Corrected the sentences.

**Pull request merged**